### PR TITLE
DAPI-1417: register 2 new key indicator related filters on the product grid

### PR DESCRIPTION
**Pull request by rouliane**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by olivierpontier — APPROVED** [review]

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Elasticsearch/Filter/KeyIndicatorFilter.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Akeneo PIM Enterprise Edition.
```
Comment (AymericPlanche) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Elasticsearch/Filter/KeyIndicatorFilter.php:6]: NOPE !

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/ProductGrid/KeyIndicatorFilter.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Akeneo PIM Enterprise Edition.
```
Comment (AymericPlanche) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/ProductGrid/KeyIndicatorFilter.php:6]: NOPE !

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/ProductGrid/RegisterKeyIndicatorFilter.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Akeneo PIM Enterprise Edition.
```
Comment (AymericPlanche) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/ProductGrid/RegisterKeyIndicatorFilter.php:6]: NOPE !

**Review by AymericPlanche — APPROVED** [review]

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Elasticsearch/Filter/KeyIndicatorFilter.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+     * {@inheritdoc}
+     */
+    public function addFieldFilter($field, $operator, $value, $locale = null, $channel = null, $options = [])
+    {
+        if (null === $this->searchQueryBuilder) {
+            throw new \LogicException('The search query builder is not initialized in the filter.');
+        }
+
+        if (!is_bool($value)) {
+            throw InvalidPropertyTypeException::arrayExpected($field, static::class, $value);
```
Comment (LaurentPetard) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Elasticsearch/Filter/KeyIndicatorFilter.php:41]:
```suggestion
            throw InvalidPropertyTypeException::booleanExpected($field, static::class, $value);
```

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Resources/config/productgrid.yml [hunk 5]
```diff
@@ -35,3 +35,55 @@ services:
...
+            - 'data_quality_insights_enrichment_quality'
+            - 'akeneo_data_quality_insights.product_grid.filter_label.enrichment_quality'
+        tags:
+            - { name: kernel.event_listener, event: oro_datagrid.datgrid.build.before.product-grid, method: buildBefore }
+
+    akeneo.pim.automation.data_quality_insights.product_grid.elasticsearch.enrichment_quality_filter:
+        class: Akeneo\Pim\Automation\DataQualityInsights\Infrastructure\Elasticsearch\Filter\KeyIndicatorFilter
+        arguments:
+            - 'data_quality_insights_enrichment_quality'
+            - 'good_enrichment'
```
Comment (LaurentPetard) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Resources/config/productgrid.yml:61]: Use constant `ProductsWithGoodEnrichment::CODE`

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Resources/config/productgrid.yml [hunk 6]
```diff
@@ -35,3 +35,55 @@ services:
...
+            - 'data_quality_insights_images_quality'
+            - 'akeneo_data_quality_insights.product_grid.filter_label.images_quality'
+        tags:
+            - { name: kernel.event_listener, event: oro_datagrid.datgrid.build.before.product-grid, method: buildBefore }
+
+    akeneo.pim.automation.data_quality_insights.product_grid.elasticsearch.image_quality_filter:
+        class: Akeneo\Pim\Automation\DataQualityInsights\Infrastructure\Elasticsearch\Filter\KeyIndicatorFilter
+        arguments:
+            - 'data_quality_insights_images_quality'
+            - 'has_image'
```
Comment (LaurentPetard) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Resources/config/productgrid.yml:87]: Use constant `ProductsWithImage::CODE`

**Review by LaurentPetard — APPROVED** [review]

**Pull request merged**